src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -393,6 +411,51 @@ public void onChange(InetAddressAndPort endpoint, ApplicationState state, Versio
                     minVersionSupplier.recompute();
             }
         });
+
+        subscribers.add(new IEndpointStateChangeSubscriber()
```

Comment:
you are dropping the event type, in cases like `onRemove` this can be an internal decision, so 0 gossip state change will be present in the row, so the only difference would be the timestamp on creating the `GossipStateTransition` and not the actual contents of the state.